### DISCUSSION TITLE
Could not find a property named 'keyName' on type 'MyClass' on ASP.NET Core OData 8.2.5

### DISCUSSION THREAD

**Discussion by abhinavkumarverma-gmail** [OData/AspNetCoreOData#1252] [General]
Assemblies affected
ASP.NET Core OData 8.2.5

Describe the bug
I am getting error - Could not find a property named 'keyName' on type 'MyClass' on ASP.NET Core OData 8.2.5, but it is working fine till ASP.NET Core OData 8.2.4 version.

The exception is raised at line "(int)oDataQueryOptions.Count?.GetEntityCount(oDataQueryOptions.Filter?.ApplyTo(myClassQuery, new ODataQuerySettings()) ?? myClassQuery)" mentioned in below code snippet

Reproduce steps

Create a class, c# WebApi as show below:
Call the API with URL as "https://localhost:8080/api/requests?$format=json&$top=10&$orderby=keyName&$count=true"
It will give an exception.
Model
Public class MyClass
{
[Key]
public int AutoId {get;set;}
public string KeyName {get;set;}
public string Description {get;set;}
}

WebAPI
public Task GetMyClassData(ODataQueryOptions oDataQueryOptions)
{
var myClassQuery = dbContext.Table1
.Select(t => new MyClass
{
AutoId = t.Autoid,
KeyName = t.KeyName,
Description = t.Description,
});

var response = new
{
totalRecords = (int)oDataQueryOptions.Count?.GetEntityCount(oDataQueryOptions.Filter?.ApplyTo(myClassQuery, new ODataQuerySettings()) ?? myClassQuery),
records = oDataQueryOptions.ApplyTo(myClassQuery).Cast()
};
}

Request/Response
Request: https://localhost:8080/api/requests?$format=json&$top=10&$orderby=keyName&$count=true)
Response: Could not find a property named 'keyName' on type 'MyClass'

Expected behavior
It should return valid object

**Comment by rbalagangadharan** [comment]
Could you please provide the response received when the $orderby parameter is not provided?

**Reply by abhinavkumarverma-gmail** [reply]
1. When $orderby parameter is not provided then List of MyClass model response is returned and api works fine.  
2. If we change the $orderby=KeyName (here the cases of KeyName mataches the model) then also it works fine. 
3. But the only issue is when i am sending the property name as camel Case i.e. $orderby=keyName (here k is in small case) then i am getting an error only in version-ASP.NET Core OData 8.2.5

**Reply by rbalagangadharan** [reply]
As per this [PR](https://github.com/OData/AspNetCoreOData/pull/1192) this should work.
